### PR TITLE
Passkeys: Retry backoff creation on domain association errors

### PR DESCRIPTION
**Pull request by nalinbhardwaj**
Tested it fixes the issue consistently, deleted app and reinstalled and immediately clicked restore from backup ~10 times.

Additionally, remove .xyz domains just in case they can cause issues.

Example log:

image

### CODE REVIEW

File: apps/daimo-mobile/src/logic/passkey.ts [hunk 1]
```diff
@@ -15,6 +15,11 @@ import { env } from "./env";
 import { Log } from "./log";
 import { parseCreateResponse, parseSignResponse } from "./passkeyParsers";
 
+function matchAASABugError(e: string) {
+  // Match without english text since iOS errors are localized to device language
+  return e.includes("JV8PYC9QV4.com.daimo") && e.includes("daimo.com");
```
Comment (dcposch) [inline apps/daimo-mobile/src/logic/passkey.ts:20]: add comment eg 
```
// Workaround iOS deeplinks bug: https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/issues/825
```

File: apps/daimo-mobile/src/logic/passkey.ts [hunk 2]
```diff
@@ -35,14 +40,17 @@ export async function createPasskey(
...
-      challengeB64,
-    })
+    () =>
+      ExpoPasskeys.createPasskey({
+        domain: env(daimoChain).passkeyDomain,
+        passkeyName,
+        passkeyDisplayTitle,
+        challengeB64,
+      }),
+    3,
```
Comment (dcposch) [inline apps/daimo-mobile/src/logic/passkey.ts:52]: 🚀

File: apps/daimo-mobile/src/view/screen/keyRotation/AddKeySlotButton.tsx [hunk 3]
```diff
@@ -78,9 +78,7 @@ export function AddKeySlotButton({
     },
   });
 
-  const didUserCancel =
-    message.includes("User cancelled") || // Android
-    message.includes("User canceled"); // iOS
+  const didUserCancel = message.includes("User cancelled");
```
Comment (dcposch) [inline apps/daimo-mobile/src/view/screen/keyRotation/AddKeySlotButton.tsx:81]: haha nice

File: apps/daimo-mobile/src/logic/log.ts [hunk 4]
```diff
@@ -41,6 +41,30 @@ export class Log {
...
+    for (let i = 0; i < retries; i++) {
+      try {
+        return await this.promise(type, fn());
+      } catch (e: any) {
+        if (matchError && matchError(getErrMessage(e))) {
+          console.log(`[LOG] ${type} trial ${i} error ${getErrMessage(e)}`);
+          await new Promise((r) => setTimeout(r, 200 * 2 ** i));
+        } else {
+          console.log(
+            `[LOG] ${type} skipping retry ${i} error ${getErrMessage(e)}`
```
Comment (dcposch) [inline apps/daimo-mobile/src/logic/log.ts:59]: nit to shorten `type} trial ${i}, quitting ...` (we don't skip, but quit trying altogether)

**Review by dcposch — APPROVED** [review]

File: apps/daimo-mobile/src/logic/log.ts (hunk 4 above)
Comment (nalinbhardwaj) [reply apps/daimo-mobile/src/logic/log.ts]: done

File: apps/daimo-mobile/src/logic/passkey.ts (hunk 1 above)
Comment (nalinbhardwaj) [reply apps/daimo-mobile/src/logic/passkey.ts:21]: added

**Pull request merged**